examples/block.rs

```diff
@@ -23,35 +15,11 @@ use ratatui::{
 // types. They are not necessary for the functionality of the code.
 type Frame<'a> = ratatui::Frame<'a, CrosstermBackend<Stdout>>;
 type Terminal = ratatui::Terminal<CrosstermBackend<Stdout>>;
-type Result<T> = std::result::Result<T, Box<dyn Error>>;
 
 fn main() -> Result<()> {
-    let mut terminal = setup_terminal()?;
-    let result = run(&mut terminal);
-    restore_terminal(terminal)?;
-
-    if let Err(err) = result {
-        eprintln!("{err:?}");
-    }
-    Ok(())
-}
-
-fn setup_terminal() -> Result<Terminal> {
-    enable_raw_mode()?;
-    let mut stdout = stdout();
-    execute!(stdout, EnterAlternateScreen)?;
-    let backend = CrosstermBackend::new(stdout);
-    let terminal = Terminal::new(backend)?;
-    Ok(terminal)
-}
-
-fn restore_terminal(mut terminal: Terminal) -> Result<()> {
-    disable_raw_mode()?;
-    execute!(terminal.backend_mut(), LeaveAlternateScreen)?;
-    Ok(())
-}
```

Comment:
Now that a restore/reset function is not needed anymore and everything is handled when the `Terminal` goes out of scope, how would that affect the backwards compatibility? In other words, do we need a major bump for these changes?

I saw that you mentioned this in the PR commments but I couldn't grasp how this is backwards compatible.

Reply:
I think we have a few breaking changes already, so that part of the question is moot.

In the initial code for this, I had it always leave alternate / disable raw mode, but changed to a flag for those because of the issues mentioned. The previous setup / restore code still works, it's just unnecessary if you create the backend using code that tracks the various things to cleanup. If you instead call crossterm::enable_raw_mode() etc. then things still behave in the same way and you need to do the normal things to cleanup.

To check for backwards compatibility, pull this PR locally and checkout the previous commit (before the changes for the examples). Every example should run fine without any changes (tested on my Mac). The ones that were most concerning were `inline` and `panic` as they're a little different from the rest. With inline I found that leaving the alternate screen cleared the text that was expected to be left. With panic, I mainly needed to find a good way to restore the terminal from in the panic hook as drop runs after the hook. Both things are fixed.